index.rst

```diff
@@ -1,61 +1,461 @@
+  };
 
-.. TODO: Delete the note below before merging new content to the master branch.
+In the example above, the ``@default`` annotation overwrites the default value of the ``MyTopic.y`` field to ``NaN``.
+If this field is omited by the CSC that publishes this topic, DDS would automatically fill in the default value ``NaN`` for the application that subscribes to it.
 
-.. note::
+By default, the default value of an enumeration corresponds to the first value in the enumeration.
+The ``@default_literal`` annotation is used to select a different value in the enumeration as the default value.
+
+.. code-block:: none
+  :emphasize-lines: 3
+
+  enum Color {
+    GREEN,
+    @default_literal RED,
+    BLUE
+  };
+
+Currently, in ``ts_xml``, it is not possible to set optional fields or explicitly set default values for fields.
+An application that subscribes to a DDS topic will get what the CSCs publish or the default values presented in Table 3.
+
+Kafka
+=====
+
+We use `Apache Avro`_ to enconde messages in Kafka.
+Avro has a primitive type ``null`` that can be used to specify optional fields in derived types like ``records``, for example:
+
+.. _`Apache Avro`: https://avro.apache.org/docs/current/spec.html
+
+.. code-block:: js
+  :emphasize-lines: 6
+
+  {
+    "type": "record",
+    "name": "foo",
+    "fields" : [
+      {"name": "bar", "type": "float"},
+      {"name": "baz", "type": ["null", "float"], "default": null} // optional field baz
+    ]
+  }
+
+However, when an Avro record is deserialized, according to `this issue`_, the JSON representation will have all the fields defined, and the optional fields will have the value ``null``.
+
+.. _this issue: https://github.com/mtth/avsc/issues/118
+
+A default value can also be specified for fields.
+In ``records`` the default value for a field can be specified as follows:
+
+.. code-block:: js
+  :emphasize-lines: 6
+
+  {
+    "type": "record",
+    "name": "foo",
+    "fields" : [
+      {"name": "bar", "type": "float"},
+      {"name": "baz", "type": "float", "default": NaN} // field baz default to NaN
+    ]
+  }
+
+From this and previous sections, SAL Kafka could, in principal, convert an IDL schema with optional and default fields into an Avro schema.
+
+A field with the value ``NaN`` in a DDS record passed along to the Avro record, and that has been tested with SAL Kafka.
+
+InfluxDB
+========
+
+InfluxDB represents a point in a time series by the `line protocol`_:
+
+.. _line protocol: https://docs.influxdata.com/influxdb/v2.0/reference/syntax/line-protocol/
+
+.. code-block:: none
+
+  <measurement>[,<tag_key>=<tag_value>[,<tag_key>=<tag_value>]] <field_key>=<field_value> [,<field_key>=<field_value>] [<timestamp>]
+
+In the EFD, a **mesasurement** corresponds to a DDS topic, and **tags** are metadata associated to **fields**. In InfluxDB, we specify a **timestamp** in the Unix epoch. If a timestamp is not specified, InfluxDB uses the server's local time with nanoseconds precision.
```

Comment:
metadata associated to --> metadata associated with
nanoseconds precision --> nanosecond precision